regions_www/m/server.R

```diff
@@ -19,804 +19,1071 @@
+  # NB: currently this resets every time scenario changes or lines get redrawn because of map zoom - possibly to revisit
+  observe({
+    input$line_type
 
-    output$download_l_csv <- downloadHandler(
-      filename = function() { "lines.csv"  },
-      content = function(file) { write.csv(signif_sdf(to_plot$straight_line)@data[codebook_l$`Variable name`], file = file) }
-    )
+    if (input$line_type == 'routes') {
+      leafletProxy("map") %>% addLayersControl(
+        position = c("bottomright"),
+        overlayGroups = c("routes_fast", "routes_quieter"),
+        options = layersControlOptions(collapsed = T)
+      )
+    } else if (input$line_type != 'routes') {
+      leafletProxy("map") %>% showGroup(c("routes_fast", "routes_quieter")) %>% removeLayersControl()
+    }
+  })
 
-    output$download_z_csv <- downloadHandler(
-      filename = function() { "zones.csv"  },
-      content = function(file) { write.csv(signif_sdf(to_plot$zones)@data[codebook_z$`Variable name`], file = file) }
-    )
+  ## Read region_stats.html, if it exists, for the loaded region
+  output$region_stats <- renderUI({
+    input$purpose
+    input$geography
+    region$current
 
-    output$download_z_geojson <- downloadHandler(
-      filename = function() { "zones.geojson"  },
-      content = function(file) { geojsonio::geojson_write(signif_sdf(to_plot$zones[codebook_z$`Variable name`]), file = file) }
-    )
+    region_stats_file <-
+      file.path("../tabs/region_stats", input$purpose, input$geography, region$current,
+                "region_stats.html")
+    if (file.exists(region_stats_file))
+      includeHTML(region_stats_file)
+    else
+      HTML("<strong>No statistics available</strong>")
+  })
 
-    output$download_l_geojson <- downloadHandler(
-      filename = function() { "lines.geojson"  },
-      content = function(file) { geojsonio::geojson_write(signif_sdf(to_plot$straight_line[codebook_l$`Variable name`]), file = file) }
-    )
+  ## Read regional data download files
+  # This file updates whenever there is a change to input$region, and content of links also depends on repo_sha
+  output$download_region_current <- renderUI({
+    region$current
 
-    output$download_rf_geojson <- downloadHandler(
-      filename = function() { "routes_fast.geojson"  },
-      content = function(file) { geojsonio::geojson_write(signif_sdf(to_plot$faster_route[codebook_r$`Variable name`]), file = file) }
-    )
+    # Create region name for files to be downloaded, so that they are accessible from the download_region.Rmd file
+    download_region <- data.frame(region_name = region$current)
+    saveRDS(download_region, file="../tabs/download_params_region_current.rds")
 
-    output$download_rq_geojson <- downloadHandler(
-      filename = function() { "routes_quiet.geojson"  },
-      content = function(file) { geojsonio::geojson_write(signif_sdf(to_plot$quieter_route[codebook_r$`Variable name`]), file = file) }
+    knitr::knit2html(quiet = T,
```

Comment:
We are `knitr` ing files on the fly?  
I'm unsure about this...
Why can't we do this so all the files have the html pre-generated

Reply:
We could have `repo_sha` in the file and gsub it.

I've not thought much about it but this worries me a bit.

Reply:
Also not sure..

Reply:
Ok, if this causes any issues I'd like to move this to static file generation please.

Reply:
switching to static would seem fine to me, and certainly if it causes any issues.  Could be moved to pct-scripts, like the region_stats files are